### PR TITLE
Generate Staged Kube Resources for Black Duck

### PR DESCRIPTION
**Pull request by mphammer**
synopsyctl generates Kube resources for specific parts of the Black Duck deployment process

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_create.go [hunk 1]
```diff
@@ -314,7 +320,24 @@ var createBlackDuckNativeCmd = &cobra.Command{
...
 		log.Debugf("generating Kubernetes resources for Black Duck '%s' in namespace '%s'...", blackDuckName, blackDuckNamespace)
-		return PrintResource(*blackDuck, nativeFormat, true)
+		app := getDefaultApp()
+		var cList *api.ComponentList
+		blackDuck.Spec.LivenessProbes = true // enable LivenessProbes when generating Kubernetes resources for customers
+		switch {
+		case !blackDuckNativeDatabase && !blackDuckNativePVC:
+			return PrintResource(*blackDuck, nativeFormat, true)
+		case blackDuckNativeDatabase:
+			cList, err = app.Blackduck().GetComponents(blackDuck, "DATABASE")
```
Comment (rrati) [inline pkg/synopsysctl/cmd_create.go]: Make these strings ("DATABASE", "PVC") constants

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#pullrequestreview-254239904" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#pullrequestreview-254243057" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#pullrequestreview-254239904" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,mphammer,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: once the present PR merges, I will cherry-pick it on top of release-2019.6.x in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/556#issuecomment-505605179):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: pkg/apps/blackduck/blackduck.go [hunk 2]
```diff
@@ -195,5 +195,14 @@ func (b Blackduck) GetComponents(bd *blackduckapi.Blackduck) (*api.ComponentList
 	if err != nil {
 		return nil, err
 	}
-	return creater.GetComponents(bd)
+	switch strings.ToUpper(compType) {
+	case "BLACKDUCK":
+		return creater.GetComponents(bd)
+	case "DATABASE":
```
Comment (rrati) [inline pkg/apps/blackduck/blackduck.go]: Use the constants here. :) . Also add BLACKDUCK as a constant

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#pullrequestreview-254243057" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/556#pullrequestreview-254691631" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,mphammer,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: new pull request created: #562

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/556#issuecomment-505605179):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**